package ProtoMsgData;
import "common.proto";

enum ServerType 
{
	 ST_Client = 0;
	 ST_Login = 1;
	 ST_Super = 2;
	 ST_Record = 3;
	 ST_Scene = 4;
	 ST_Gateway = 5;
}

enum ServerStatus 
{
	 ST_Wait = 1;
	 ST_Start = 2;
	 ST_Close = 3;
}

message ReqServerInfo
{
	 required ServerType servertype = 1;
}

message ServerInfo
{
	 optional ServerType servertype = 1;
	 optional uint32 id = 2;
	 optional string ip = 3 [default = ""];
	 optional uint32 port = 4 [default = 0];
	 optional ServerStatus status = 5 [default = ST_Wait];
}

message AckUpdateServerInfo
{
	 optional ServerInfo serverinfo = 1;
}

message AckRespServerInfo
{
	 optional uint32 id = 1 [default = 0];
}

message AckNotifyMe
{
	 repeated ServerInfo serverinfo = 1;
}

message ReqHeartBeat
{
}

message AckHeartBeat
{
}

message ReqVerifyServer
{
	 optional ServerInfo serverinfo = 1;
}

message AckVerifyServer
{
	 optional bool ret = 1;
}

enum GatewayStatus 
{
	 GS_Close = 0;
	 GS_Normal = 1;
	 GS_Full = 2;
}

message GatewayInfo
{
	 optional uint32 id = 1;
	 optional string ip = 2;
	 optional uint32 port = 3;
	 optional uint32 person = 4;
	 optional GatewayStatus status = 5;
}

message ReqCreateUser
{
	 optional string phone = 1;
	 optional uint64 id = 2;
}

message AckCreateUser
{
	 optional ErrorCode code = 1;
	 optional string phone = 2;
	 optional uint64 id = 3;
	 optional uint64 charid = 4;
}
